plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -380,6 +380,14 @@ void state_history_plugin_impl::plugin_shutdown() {
    accepted_block_connection.reset();
    block_start_connection.reset();
    thread_pool.stop();
+
+   // This is a temporary fix until https://github.com/AntelopeIO/spring/issues/842 can be worked.
+   // Drain the io_service of anything that could be referencing state_history_plugin
+   if (app().executor().get_io_service().stopped()) {
+      app().executor().get_io_service().restart();
+      while (app().executor().get_io_service().poll())
+         ;
+   }
```

Comment:
It looks like a SIGINT etc here will interrupt it. Probably fine.

I wonder if there is a way to add some sort of sentinel to know that "anything that could be referencing state_history_plugin" _has_ actually been drained, so we don't somehow get in a state where we loop forever here because something else unrelated is `post()`ing forever. I'm not immediately coming up with a good solution though.

Reply:
Maybe just not do the `poll()` call in a loop. We are not expecting anything to be posting at this point.

Reply:
Interesting enough, the `poll()` has to be in a loop. I guess the `co_spawn` must be posting additional as it drains.